plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java

```diff
@@ -259,18 +275,39 @@ private Instance createInstance(String imageId,
                                     String machineId,
                                     String creator,
                                     String workspaceId,
-                                    boolean bindWorkspace,
+                                    boolean isDev,
                                     String displayName,
                                     Recipe recipe,
                                     int memorySizeMB,
                                     LineConsumer outputConsumer)
             throws MachineException {
         try {
+            final Map<String, String> labels;
+            final Map<String, Map<String, String>> portsToExpose;
+            final Set<String> volumes;
+            if (isDev) {
+                labels = new HashMap<>(machineContainerLabels.size() + devMachineContainerLabels.size());
```

Comment:
What do you think about <code>Maps.newHashMapWithExpectedSize(...)</code> in this case?